tests/integration/helpers.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import tempfile
+from typing import Optional
+
+PERMISSION_MASK_FOR_SCP = 644
+
+
+async def pull_content_from_unit_file(unit, path: str) -> str:
+    """Pull the content of a file from one unit.
+
+    Args:
+        unit: the Juju unit instance.
+        path: the path of the file to get the contents from.
+
+    Returns:
+        the entire content of the file.
+    """
+    # Get the file's original access permission mask.
+    result = await run_command_on_unit(unit, f"stat -c %a {path}")
+    permissions = result.strip()
+    # Change the permission in order to be able to retrieve the file using the ubuntu user.
+    if permissions != PERMISSION_MASK_FOR_SCP:
+        await run_command_on_unit(unit, f"chmod {PERMISSION_MASK_FOR_SCP} {path}")
+
+    # Get the contents of the file.
+    temp_file = tempfile.NamedTemporaryFile()
+    await unit.scp_from(path, temp_file.name, scp_opts=["-v"])
+    data = temp_file.read().decode("utf-8")
+    temp_file.close()
+
+    # Change the file permissions back to the original mask.
+    if permissions != PERMISSION_MASK_FOR_SCP:
```

Comment:
This condition doesn't seem right? I think you need a flag somewhere to show that you modified the permissions just after line 26 above then check for that condition here?

Reply:
Sounds good to me, updating this

@marceloneppel , tagging you here since I used your code for this and we should keep it consistent. Maybe we should add this to a library instead since it seems quite useful and I could see other people needing :-)

Reply:
Thanks Mia! Yes, I agree that makes sense to create a library for that. We can discuss about it as probably there will be more charms testing this conditions.